trieste/space.py

```diff
@@ -1136,6 +1336,23 @@ def product(self, other: TaggedProductSearchSpace) -> TaggedProductSearchSpace:
         """
         return TaggedProductSearchSpace(spaces=[self, other])
 
+    @property
+    def one_hot_encoder(self) -> EncoderFunction:
+        """An encoder that one-hot-encodes all subpsaces that support it (and leaves
+        the other subspaces unchanged)."""
+
+        def encoder(x: TensorType) -> TensorType:
+            components = []
+            for tag in self.subspace_tags:
+                component = self.get_subspace_component(tag, x)
+                space = self.get_subspace(tag)
+                if isinstance(space, HasOneHotEncoder):
+                    component = space.one_hot_encoder(component)
+                components.append(component)
+            return tf.concat(components, axis=1)
```

Comment:
Nitpick: suggest using `axis=-1`, allowing an arbitrary number of batch dimensions.